build-support/docker/centos6/Dockerfile

```diff
@@ -6,16 +6,27 @@ FROM centos:6
 
 # Install python 2.7.13, (more) modern gcc, and a JDK.
 RUN yum -y update
+# TODO: figure out why this needs to be installed first for /usr/bin/scl to work!
 RUN yum install -y centos-release-scl
-RUN yum install -y \
-  devtoolset-7-gcc \
-  devtoolset-7-gcc-c++ \
+RUN yum install -y devtoolset-7-gcc{,-c++} \
   git \
   java-1.8.0-openjdk-devel \
   python27 \
-  libffi \
-  libffi-devel \
-  openssl-devel
+  libffi{,-devel} \
+  openssl-devel \
+  sqlite-devel
+
+# Install the desired version of python 3.
+ARG PYTHON_3_VERSION=3.6.8
+# NB: PYENV_ROOT must be set in the environment for `pyenv install` to work, otherwise it will
+# mysteriously fail to find the `install` command.
+ENV PYENV_ROOT /pyenv-py3-docker-build
+RUN mkdir ${PYENV_ROOT}
+RUN git clone https://github.com/pyenv/pyenv ${PYENV_ROOT}
+RUN /usr/bin/scl enable devtoolset-7 -- bash -c '\
+    ${PYENV_ROOT}/bin/pyenv install --keep --skip-existing ${PYTHON_3_VERSION} \
+    && ${PYENV_ROOT}/bin/pyenv global ${PYTHON_3_VERSION}'
```

Comment:
I realized I don't think we want this line. This will override the global `python` bin to point to pyenv's py3.6.8. All we want is to install python 3.6, not to set it as default (which is a different objective than we have in Pex's travis.yml).